Arena

premkumr opened the pull request.

premkumr commented:
@daverigby - I've set the `opt.narenas` to 1 & added more arenas using `arenas.extend`.  Hopefully we dont have to worry about those being multiplexed by je_malloc itself. or else, we will have to modify our `platform_thread_wrap`.

daverigby commented on daemon/alloc_hooks.def:
```diff
@@ -32,3 +32,5 @@ X(release_free_memory, void, ())
 X(get_allocator_property, bool, (const char* name, size_t* value))
 X(set_allocator_property, bool, (const char* name, size_t value))
+X(set_allocator_arena, bool, (size_t arenaid))
+X(get_arena_allocation_size, size_t, (size_t arenaid))
```
I'm not sure if we want to add two new methods for these, or simply use the existing get/set_allocator_property APIs - while jemalloc is the only "production" allocator we currently use, we do run tests with the system allocator and badmalloc - and hence it would confuse the interface if we add a bunch of methods specific to jemalloc, when we can just use the generic method.

daverigby commented on daemon/alloc_hooks_jemalloc.cc:
```diff
@@ -44,6 +44,38 @@ static int jemalloc_get_stats_prop(const char* property, size_t* value) {
+    T v;
+    size_t miblen = sizeof(mib) / sizeof(size_t);
+    size_t sz = sizeof(T);
+    je_mallctlnametomib(propname, mib, &miblen);
```
Given the MIB lookup is supposed to be a bit slow, for production use (when associating threads  with arenas) we  want to cache the MIB lookup (nametomib) result and call je_mallctlbymib() directly.

maniktaneja committed:
Add RBAC with ssl client certificate for memcached

This commits add the support for extracting username
from the ssl client cerificate and use it for role
base authentication.

Following new parameters are added in memcached config:

1. path - It is the field which will be used to extract the
          username. Currently few fields in common name and
          subject alternate name are supported

2. prefix - Prefix string to be ignored from the path value

3. delimiter - It can contains multiple character and username
               will be truncated till the offset of any of the
               character in delimiter is found

Change-Id: I91c5591a7d154ab28a4cffcc22e846744d1a98e6
Reviewed-on: http://review.couchbase.org/75765
Tested-by: Build Bot <build@couchbase.com>
Reviewed-by: Dave Rigby <daver@couchbase.com>
Reviewed-by: Trond Norbye <trond.norbye@gmail.com>

trondn committed:
Generate new certificates

Add a new certificate containing alternative names, and
set the numbers of bits to 1k (smaller and faster)

Change-Id: I2342425e92124fbaaf97d21351c8e395f95a61f6
Reviewed-on: http://review.couchbase.org/76410
Reviewed-by: Manik Taneja <manik.taneja@gmail.com>
Tested-by: Build Bot <build@couchbase.com>
Reviewed-by: Dave Rigby <daver@couchbase.com>

Commit:
support for arenas

Change-Id: I638950af897ccaf53c33641c30e49a136febabf5

trondn committed:
Fixup copy'n'paste bug installing key files

Change-Id: I78012f960add2b07688d6305444d3f0e8dc3c4b6
Reviewed-on: http://review.couchbase.org/76459
Tested-by: Build Bot <build@couchbase.com>
Reviewed-by: Trond Norbye <trond.norbye@gmail.com>

trondn committed:
Add unique_ptr for X509 ptr in cb::openssl

Change-Id: I5d9e31c44db28330032a0945b0b9cebef358e7b0
Reviewed-on: http://review.couchbase.org/76616
Reviewed-by: Dave Rigby <daver@couchbase.com>
Tested-by: Build Bot <build@couchbase.com>

Commit:
thread.arena mib

Change-Id: Iec3ab250ba0ac925fb07c42253c22f473febe10e
